app/controllers/concerns/cache_concern.rb

```diff
@@ -28,7 +28,7 @@ def enforce_cache_control!
   def render_with_cache(**options)
     raise ArgumentError, 'Only JSON render calls are supported' unless options.key?(:json) || block_given?
 
-    key        = options.delete(:key) || [[params[:controller], params[:action]].join('/'), options[:json].respond_to?(:cache_key) ? options[:json].cache_key : nil, options[:fields].nil? ? nil : options[:fields].join(',')].compact.join(':')
+    key        = options.delete(:key) || [[params[:controller], params[:action]].join('/'), options[:json].respond_to?(:cache_key) ? options[:json].cache_key : nil, options[:fields]&.join(',')].compact.join(':')
```

Comment:
I think this is safe by itself, but also the `compact` on the end.

The entire second half of this method (how to handle when `options[:key]` isnt present) - should be a separate method.